nlp_apps.ipynb

```diff
@@ -797,11 +797,215 @@
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {
     "collapsed": true
    },
+   "source": [
+    "## Text Analysis using TF-IDF\n",
+    "Since we know that computers are great with numbers but they cannot work out with the natural language. So in-order to overcome that text can be directly converted to numbers for analyzing.One of the most common and popular technique for this is TF-IDF which stands for Term Frequency and Inverse Document Frequency. \n",
+    "\n",
+    "1. **TF(Term -Frequency):-**Gives the frequency of each word in a document. As the number of occurances of a word increases in a document its value increases for that document. Basically, if a word appears more times in a docuemnt then that word is important to that document. \n",
+    "    \n",
+    "    **TF(t) = (Number of times term t appears in a document) / (Total number of terms in the document)**\n",
+    "\n",
+    "\n",
+    "2. **IDF(Inverse Document Frequency):-** It calculates, how much a word is important for a given document. The words that occur in less documents are more important as compared to the words that occur in more number of documents. It helps to find the important words across the documents .\n",
+    "\n",
+    "    **IDF(t) = log(Total number of documents / (1+Number of documents with term t in it))**\n",
+    "    \n",
+    "    (1 is added to the base for coding purposes. It helps to deal with the cases when the term does not appears in any document)\n",
+    "\n",
+    "\n",
+    "*TF-IDF* finally gives the importance to a single word in a collection of documents by multiplying the TF of that word in that document with the  IDF of that word  across the documents.\n",
+    "\n",
+    "\n",
+    "#### Working with TF-IDF  \n",
+    "1. First of all we need a list of all the documents to work on.\n",
+    "2. Next, basic text-procesing is done over all the documents.\n",
+    "3. Now, an object is made of the class tf_idf_analyis with the list of all pre-processed text.\n",
+    "4. We can find the term-frequency of a given document using *get_term_frequency(doc)*.\n",
+    "5. The relevance of word can be calculated using *get_tf_idf_score(word,doc)* function. \n",
+    "6. Using this object we can get the *n* most important words of the document using *top(doc,n)* function.\n",
+    "7. We can see the tf-idf vector i.e. score of all the words in the document by *tf_idf_vector(doc)*.\n",
```

Comment:
Add commas around the 'i.e. score of all the words in the document' part. Remember tildes.